packages/flutter/lib/src/material/arc.dart

```diff
@@ -0,0 +1,174 @@
+      }
+    }
+  }
+
+  Point _center;
+  double _radius;
+  double _beginAngle;
+  double _endAngle;
+
+  @override
+  Point transform(double t) {
+    if (_beginAngle == null || _endAngle == null)
+      return Point.lerp(begin, end, t);
+    final double angle = lerpDouble(_beginAngle, _endAngle, t);
+    final double x = math.cos(angle) * _radius;
+    final double y = math.sin(angle) * _radius;
+    return  _center + new Offset(x, y);
+  }
+}
+
+enum _CornerId {
+  topLeft,
+  topRight,
+  bottomLeft,
+  bottomRight
+}
+
+class _Diagonal {
+  const _Diagonal(this.beginId, this.endId);
+  final _CornerId beginId;
+  final _CornerId endId;
+}
+
+class MaterialRectArc extends _MaterialArc<Rect> {
+  static final List<_Diagonal> _allDiagonals = <_Diagonal>[
+    const _Diagonal(_CornerId.topLeft, _CornerId.bottomRight),
+    const _Diagonal(_CornerId.bottomRight, _CornerId.topLeft),
+    const _Diagonal(_CornerId.topRight, _CornerId.bottomLeft),
+    const _Diagonal(_CornerId.bottomLeft, _CornerId.topRight),
+  ];
+
+  MaterialRectArc({
+    @required Rect begin,
+    @required Rect end
+  }) : super(begin: begin, end: end) {
+    final Offset centersVector = end.center - begin.center;
+    double maxSupport = 0.0;
+    for (_Diagonal diagonal in _allDiagonals) {
+      final double support = _diagonalSupport(centersVector, diagonal);
+      if (support > maxSupport) {
+        _diagonal = diagonal;
+        maxSupport = support;
+      }
+    }
+    _beginArc = new MaterialPointArc(
+      begin: _cornerFor(begin, _diagonal.beginId),
+      end: _cornerFor(end, _diagonal.beginId)
+    );
+    _endArc = new MaterialPointArc(
+      begin: _cornerFor(begin, _diagonal.endId),
+      end: _cornerFor(end, _diagonal.endId)
+    );
+  }
+
+  _Diagonal _diagonal;
+  MaterialPointArc _beginArc;
+  MaterialPointArc _endArc;
+
+  Point _cornerFor(Rect rect, _CornerId id) {
+    switch (id) {
+      case _CornerId.topLeft: return rect.topLeft;
+      case _CornerId.topRight: return rect.topRight;
+      case _CornerId.bottomLeft: return rect.bottomLeft;
+      case _CornerId.bottomRight: return rect.bottomRight;
+    }
+    return Point.origin;
+  }
+
+  double _diagonalSupport(Offset centersVector, _Diagonal diagonal) {
+    final Offset delta = _cornerFor(begin, diagonal.endId) - _cornerFor(begin, diagonal.beginId);
+    final double length = delta.distance;
+    return centersVector.dx * delta.dx / length + centersVector.dy * delta.dy / length;
+  }
+
+  @override
+  Rect transform(double t) {
+    Point beginArcPoint = _beginArc.transform(t);
+    Point endArcPoint = _endArc.transform(t);
+    double minX = math.min(beginArcPoint.x, endArcPoint.x);
+    double maxX = math.max(beginArcPoint.x, endArcPoint.x);
+    double minY = math.min(beginArcPoint.y, endArcPoint.y);
+    double maxY = math.max(beginArcPoint.y, endArcPoint.y);
+    return new Rect.fromLTRB(minX, minY, maxX, maxY);
+  }
+}
+
+class MaterialPointArcTween extends Tween<Point> {
```

Comment:
These tweens are kind of strange because you can set the points but that doesn't change the lerp.  I think it would make more sense to either (1) recompute the arc when the begin/end is set or (2) throw an exception in the setters.  I'd probably start with (2) because (1) is slightly tricky if you don't want to recompute the arc twice when begin and end are set in sequence.

In either case, we should merge the tweens into the MaterialPointArcTween/MaterialRectArc.